data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/server/PeerForwarderServerProxyTest.java

```diff
@@ -58,11 +71,22 @@ void stop_should_stop_server_if_server_started() throws ExecutionException, Inte
         when(server.start()).thenReturn(completableFuture);
         when(server.stop()).thenReturn(completableFuture);
         when(completableFuture.get()).thenReturn(mock(Void.class));
+        when(peerForwarderProvider.isPeerForwardingRequired()).thenReturn(true);
 
         final PeerForwarderServerProxy objectUnderTest = createObjectUnderTest();
         objectUnderTest.start();
         objectUnderTest.stop();
         verify(server).stop();
     }
 
+    @Test
+    void stop_should_do_nothing_if_noop_server_is_started() {
```

Comment:
This test name is confusing. Could you explain what it is testing? Looks like server should not be started or stopped since isPeerForwardingRequired() returns false. Maybe `no_server_interaction_if_peer_forwarding_not_required`?

Reply:
Yeah, it's a bit confusing. peerForwarderProvider.isPeerForwardingRequired() return false so it'll create a `NoOpPeerForwarderServer`. I can rename the test